### PR TITLE
[Wallet] Add new onboarding Welcome screen

### LINKED ISSUE #4401: [Wallet] Update onboarding changes
### Expected Behavior

Finish onboarding changes, order of the screens, and new screen to choose between sign up / restore

### Current Behavior

What is the current behavior?

### PR DESCRIPTION
**Pull request by jeanregisser**
### Description

This PR implements the new [onboarding welcome screen](https://www.figma.com/file/zt7aTQ5wuXycIwxq5oAmF9/Wallet--Refresh?node-id=6184%3A72673).

Highlights:
- This new screen is pushed after the onboarding education screen.
- Whether the user chooses `Create Account` or `Restore Account` the next few screens will be the same
  - `RegulatoryTerms`
  - `NameAndNumber`
  - `PincodeSet`
- Then the next screen will be either `EnterInviteCode` or `ImportWallet` based on the initial choice.
- Going back from those 2 screens will directly go back to the `Welcome` screen (skipping the intermediary screens.
- Making a new choice on the Welcome screen will again go through the same screens listed above, but with some info pre-filled (name and number). The PIN will need to be created again. This was decided to avoid confusion.
- The user can interrupt this onboarding process anytime. It will pick up at the appropriate place when the app is resumed/relaunched. Note that if the user didn't reach `EnterInviteCode` or `ImportWallet`, it will resume at the onboarding education.
- Add new analytics events.

See assets diff: https://github.com/clabs-co/valora-app-branding/compare/a3f5867...d58569f

### Other changes

- Renamed `JoinCelo` -> `NameAndNumber`
- Removed restore account button from `EnterInviteCode` screen, since we now have the welcome screen for that choice.
- Don't hide Android emulator window during e2e test when NOT running on the CI

### Related issues

- Fixes #4401

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/mobile/src/onboarding/registration/NameAndNumber.tsx [hunk 1]
```diff
@@ -96,7 +91,7 @@ function getPhoneNumberState(
   }
 }
 
-export class JoinCelo extends React.Component<Props, State> {
+export class NameAndNumber extends React.Component<Props, State> {
```
Comment (gnardini) [inline packages/mobile/src/onboarding/registration/NameAndNumber.tsx:94]: Love the name change, it's much clearer

**Review by gnardini — APPROVED** [review]
Looks great!

File: packages/mobile/src/navigator/Navigator.tsx [hunk 2]
```diff
@@ -475,14 +483,11 @@ export function MainStackScreen() {
...
+    } else if (!e164Number || !acceptedTerms || pincodeType === PincodeType.Unset) {
+      // User didn't go far enough in onboarding, start again from education
       initialRoute = Screens.OnboardingEducationScreen
-    } else if (!acceptedTerms) {
-      initialRoute = Screens.RegulatoryTerms
-    } else if (pincodeType === PincodeType.Unset) {
-      initialRoute = Screens.PincodeSet
     } else if (!redeemComplete && !account) {
-      initialRoute = Screens.EnterInviteCode
+      initialRoute = isRestoringAccount ? Screens.ImportWallet : Screens.EnterInviteCode
```
Comment (jmrossy) [inline packages/mobile/src/navigator/Navigator.tsx]: I would have guessed that `isRestoringAccount` is for a progress indicator. Maybe `choseToRestoreAccount`?

File: packages/mobile/test/schemas.ts [hunk 3]
```diff
@@ -335,9 +335,14 @@ export const v2Schema = {
     ...v1Schema.app,
     sessionId: '',
   },
+}
+
+export const v3Schema = {
+  ...v2Schema,
   account: {
     ...v1Schema.account,
```
Comment (jmrossy) [inline packages/mobile/test/schemas.ts]: This and line 348 should be v2 right?

**Review by jmrossy — APPROVED** [review]
Nice work! Just two minor things

File: packages/mobile/src/navigator/Navigator.tsx (hunk 2 above)
Comment (jeanregisser) [reply packages/mobile/src/navigator/Navigator.tsx]: Yes, thanks for the suggestion.

File: packages/mobile/test/schemas.ts (hunk 3 above)
Comment (jeanregisser) [reply packages/mobile/test/schemas.ts]: Oops, good catch!

**Pull request merged**